lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -70,9 +70,11 @@
 from collections.abc import Mapping
 from datetime import datetime, timedelta, timezone
 from enum import Enum
-from subprocess import CalledProcessError, CompletedProcess
+from subprocess import CalledProcessError, CompletedProcess, TimeoutExpired
 from typing import Any, Dict, Iterable, List, Optional, Union
 
+from tenacity import before_log, retry, stop_after_attempt, wait_fixed
```

Comment:
I don't think we (normally) want to introduce a dependency on a 3rd party library inside a charm lib, right? If we do, in addition to adding the charm lib, you'd have to manually add the dependency to the charm's requirements.txt to satisfy it. Doesn't seem worth it for something as straight-forward as a retry.

Reply:
That makes sense @benhoyt  :)